<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AlertSettingsDialog</class>
 <widget class="QDialog" name="AlertSettingsDialog">
  <property name="windowTitle">
   <string>Alert Settings</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/icons/settings.png</normaloff>:/icons/settings.png</iconset>
  </property>
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabThresholds">
      <attribute name="title">
       <string>Thresholds</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Alert Thresholds</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>High Bandwidth Threshold (KB/s):</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="spinBandwidthThreshold">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
            <property name="suffix">
             <string> KB/s</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Connection Spike Window (seconds):</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="spinConnectionWindow">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>3600</number>
            </property>
            <property name="suffix">
             <string> sec</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Connection Spike Threshold:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="spinConnectionThreshold">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>10000</number>
            </property>
            <property name="suffix">
             <string> connections</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Data Exfiltration Threshold (KB):</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="spinDataExfiltration">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1048576</number>
            </property>
            <property name="suffix">
             <string> KB</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabAlertTypes">
      <attribute name="title">
       <string>Alert Types</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Enable/Disable Alert Types</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QCheckBox" name="chkNewApp">
            <property name="text">
             <string>New Application Detected</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QCheckBox" name="chkHighBandwidth">
            <property name="text">
             <string>High Bandwidth Usage</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QCheckBox" name="chkSuspiciousConn">
            <property name="text">
             <string>Suspicious Connections</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QCheckBox" name="chkPortScan">
            <property name="text">
             <string>Port Scan Detected</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QCheckBox" name="chkDataExfiltration">
            <property name="text">
             <string>Data Exfiltration</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="chkProtocolAnomaly">
            <property name="text">
             <string>Protocol Anomaly</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QCheckBox" name="chkConnectionSpike">
            <property name="text">
             <string>Connection Spike</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QCheckBox" name="chkRuleViolation">
            <property name="text">
             <string>Rule Violation</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QCheckBox" name="chkCustomAlert">
            <property name="text">
             <string>Custom Alerts</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok|QDialogButtonBox::RestoreDefaults</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>AlertSettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>AlertSettingsDialog</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>spinBandwidthThreshold</tabstop>
  <tabstop>spinConnectionWindow</tabstop>
  <tabstop>spinConnectionThreshold</tabstop>
  <tabstop>spinDataExfiltration</tabstop>
  <tabstop>chkNewApp</tabstop>
  <tabstop>chkHighBandwidth</tabstop>
  <tabstop>chkSuspiciousConn</tabstop>
  <tabstop>chkPortScan</tabstop>
  <tabstop>chkDataExfiltration</tabstop>
  <tabstop>chkProtocolAnomaly</tabstop>
  <tabstop>chkConnectionSpike</tabstop>
  <tabstop>chkRuleViolation</tabstop>
  <tabstop>chkCustomAlert</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
</ui>
